src/Sarif.Viewer.VisualStudio/VersionControlParser/VersionControlParserFactory.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.RegularExpressions;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.Sarif.Viewer.Sarif;
+
+namespace Microsoft.Sarif.Viewer
+{
+    internal class VersionControlParserFactory
+    {
+        // known version control host
+        internal static string GithubHost = "github.com";
+        internal static string AdoHost = "dev.azure.com";
+
+        internal static Regex githubRegx = new Regex(@"^https?://github\.com", RegexOptions.IgnoreCase | RegexOptions.Compiled);
```

Comment:
>githubRegx [](http://example.com/codeflow?start=30&length=10)

since you are not going to share besides this class:
1. make it private
2. for static, we normally use s_githubRegex (add the 'e'). Also apply the same thing to the s_adoRegex